neet/boolean/logicnetwork.py

```diff
@@ -336,9 +343,94 @@ def read(cls, table_file):
                         raise FormatError(
                             "node must be specified before logic conditions")
 
-        if not all(table):
-            raise FormatError(
-                "logic conditions must be provided for all nodes")
+        # If no truth table is provided for a node, that node is considered
+        # an "external" node, i.e, its state stays on or off by itself.
+        for i, sub_table in enumerate(table):
+            if not sub_table:  # Empty truth table.
+                table[i] = ((i), {'1'})
+        return cls(table, names)
+
+    @classmethod
+    def read_logic(cls, logic_file, external_nodes_file=None):
+        """
+        Read a network from a file of logic equations.
+
+        A logic equations has the form of `A = B AND ( C OR D )`, each term
+        being separated from paranthesis and logic operators with at least a
+        space. The optional `external_nodes_file` takes a file that contains
+        nodes in a column whose states do not depend on any nodes. These are
+        considered "external" nodes. Equivalently, such a node would have a
+        logic equation `A = A`, for its state stays on or off unless being set
+        externally, but now the node had to be excluded from `external_nodes_file`
+        to avoid duplication and confusion.
+
+        :param logic_file: a .txt file of logic equations
+        :param external_nodes_file: a .txt file of external nodes
+        :returns: a :class:LogicNetwork
+
+        .. rubric:: Basic Use:
+
+        ::
+
+            >>> net = LogicNetwork.read_logic('myeloid-logic.txt')
```

Comment:
The docstrings here are independent of the data we use in the `data` folder. This line won't read the data even if the names match because of different paths. We do not dictate how users name their files and I used a different name to avoid the impression that this is the only acceptable file name. Did it instead introduce more confusion?